bgc_data/bgc_larval_fish_count_raw_data.sql

```diff
@@ -0,0 +1,18 @@
+-- Materialized view for larval fish count raw product
+-- To be served as a WFS layer by Geoserver using output format csv-with-metadata-header,
+-- which will convert the jsonb `abundances` column into separate CSV columns on output.
+CREATE MATERIALIZED VIEW bgc_larval_fish_count_raw AS
+WITH pivoted AS (
+   SELECT 
+      i_sample_id, 
+      trip_code,
+      jsonb_object_agg(scientificname, taxon_count) AS abundances
+   FROM bgc_lfish_countraw
+   GROUP BY trip_code, i_sample_id 
+)
+-- join on to metadata columns, include a row for every trip with larval fish samples taken
+   SELECT 
+      m.*,
+      p.abundances
+   FROM bgc_larval_fish_map m LEFT JOIN pivoted p USING (trip_code, i_sample_id)
```

Comment:
`bgc_lfish_samples` has `i_sample_id` as its [primary key](https://github.com/aodn/public-schema/blob/d1dd1abf64beb570e72d9139ffe3a13184d3f1c6/bgc_data/bgc_lfish_samples.dataresource.yaml#L92), so that fields is all you need to join here. The `trip_code` is redundant.
For the same reason, you only need to group by `i_sample_id` in the pivoted sub-query, above.

Reply:
It makes sense, I understand. 
Updated in the next commit.